src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -344,9 +350,35 @@ async def __mount(self, pool, uuid, path=SYSDATASET_PATH):
                 os.mkdir(mountpoint)
             await run('mount', '-t', 'zfs', dataset, mountpoint, check=True)
 
+        if osc.IS_LINUX:
+
+            # make sure the glustereventsd webhook dir and
+            # config file exist and start the glustereventsd
+            # service (if appropriate)
+            eventsd = await self.middleware.call('gluster.eventsd.init')
+            await eventsd.wait()
+            if eventsd.error:
+                raise CallError(f'{eventsd.error}')
+
+            # mount the local glusterfuse mount after the
+            # zfs dataset is mounted
+            ctdb = await self.middleware.call('ctdb.shared.volume.mount')
+            await ctdb.wait()
+            if ctdb.error:
+                raise CallError(f'{ctdb.error}')
+
     async def __umount(self, pool, uuid):
+
         for dataset, name in reversed(self.__get_datasets(pool, uuid)):
             try:
+                # if this is the ctdb dataset, then we have to
+                # unmount the local glusterfuse mount first before
+                # unmounting the underlying zfs dataset
+                if osc.IS_LINUX and name == CTDBConfig.CTDB_VOL_NAME.value:
+                    c = await self.middleware.call('ctdb.shared.volume.umount')
+                    await c.wait()
+                    if c.error:
+                        raise CallError(f'{c.error}')
```

Comment:
Just error may give no context, we may say "Failed to umount ctdb shared volume: {e}"

Reply:
I've actually changed it so that it's logging an error and continuing on so we don't block other mount operations in the sysdataset.setup method.